src/components/gdprsupport.main.jsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright © 2018 Elastic Path Software Inc. All rights reserved.
+ *
+ * This is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+
+import React from 'react';
+import intl from 'react-intl-universal';
+import { withRouter } from 'react-router-dom';
+import Modal from 'react-responsive-modal';
+import './gdprsupport.main.less';
+
+const Config = require('Config');
+
+class GdprSupportModal extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      open: true,
+      checked: false,
+    };
+    this.handleOpenModal = this.handleOpenModal.bind(this);
+    this.handleCloseModal = this.handleCloseModal.bind(this);
+    this.handleCheck = this.handleCheck.bind(this);
+    this.handleAcceptGdpr = this.handleAcceptGdpr.bind(this);
+  }
+
+  handleOpenModal() {
+    this.setState({ open: true });
+  }
+
+  handleCloseModal() {
+    this.setState({ open: false });
+  }
+
+  handleCheck() {
+    const { checked } = this.state;
+    this.setState({ checked: !checked });
+  }
+
+  handleAcceptGdpr() {
```

Comment:
Similarly we should handle if they "decline" the GDPR dialog. We shouldn't display the modal again on the homepage for them (unless of course they clear local storage) and should still prevent the Analytics requests.